python/src/equistore/operations/equal_metadata.py

```diff
@@ -129,6 +129,86 @@ def equal_metadata(
+) -> bool:
+    """
+    Checks if two :py:class:`TensorBlock` objects have the same metadata,
+    returning a bool.
+
+    If `check` is none (the default), all metadata (i.e. the samples,
+    components, and properties of each block) is checked to contain the same
+    values in the same order.
+
+    Passing `check` as a list of strings will only check the metadata specified.
+    Allowed values to pass are "samples", "components", and "properties".
+
+    :param block_1: The first :py:class:`TensorBlock`.
+    :param block_2: The second :py:class:`TensorBlock` to compare to the first.
+    :param check: A list of strings specifying which metadata of each block to
+        check. If none, all metadata is checked. Allowed values are "samples",
+        "components", and "properties".
+
+    :return: True if the metadata of the two :py:class:`TensorBlock` objects are
+        equal, False otherwise.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import equistore
+    >>> from equistore import Labels, TensorBlock
+    >>> block_1 = TensorBlock(
+    ...     values=np.full((4, 3, 1), 4.0),
+    ...     samples=Labels(["samples"], np.array([[0], [1], [4], [5]])),
+    ...     components=[Labels.arange("components", 3)],
+    ...     properties=Labels(["p_1", "p_2"], np.array([[0, 1]])),
+    ... )
+    >>> block_2 = TensorBlock(
+    ...     values=np.full((4, 3, 1), 4.0),
+    ...     samples=Labels(["samples"], np.array([[0], [1], [4], [5]])),
+    ...     components=[Labels.arange("components", 3)],
+    ...     properties=Labels(["p_3", "p_4"], np.array([[0, 1]])),
+    ... )
+    >>> equal_metadata_block(block_1, block_2)
+    False
+    >>> equal_metadata_block(
+    ...     block_1,
+    ...     block_2,
+    ...     check=["samples", "components"],
+    ... )
+    True
+    """
+    # Check input args
+    if not isinstance(block_1, TensorBlock):
+        raise TypeError(f"`block_1` must be a TensorBlock, not {type(block_1)}")
+    if not isinstance(block_2, TensorBlock):
+        raise TypeError(f"`block_2` must be a TensorBlock, not {type(block_2)}")
+    if not isinstance(check, (list, type(None))):
+        raise TypeError(f"`check` must be a list, not {type(check)}")
+    if check is None:
+        check = ["samples", "components", "properties"]
+    for metadata in check:
+        if not isinstance(metadata, str):
+            raise TypeError(
+                f"`check` must be a list of strings, got list of {type(metadata)}"
+            )
+        if metadata not in ["samples", "components", "properties"]:
+            raise ValueError(f"Invalid metadata to check: {metadata}")
+
+    # Check metadata of the blocks
+    try:
+        _check_blocks(block_1, block_2, check, "equal_metadata")
+    except ValueError:
+        return False
+
+    # Check metadata of the gradients
+    try:
+        _check_same_gradients(block_1, block_2, check, "equal_metadata")
+    except ValueError:
```

Comment:
I don't love that we still use ValueError for the `_check` functions, since that could hide issues in the same way as #175. Should we also use `NotEqualError` for these?